sql/src/main/java/org/apache/druid/sql/calcite/rel/Windowing.java

```diff
@@ -358,12 +359,43 @@ public WindowFrame getWindowFrame()
       if (group.lowerBound.isUnbounded() && group.upperBound.isUnbounded()) {
         return WindowFrame.unbounded();
       }
+      int lowerOffset = figureOutOffset(group.lowerBound);
+      int upperOffset = figureOutOffset(group.upperBound);
+      if (group.lowerBound.isPreceding()) {
+        lowerOffset = -lowerOffset;
+      }
+      if (group.upperBound.isPreceding()) {
+        upperOffset = -upperOffset;
+      }
+      if (lowerOffset >= upperOffset) {
+        final String first, second;
+        int val;
+        if (group.lowerBound.isPreceding()) {
+          val = Math.abs(lowerOffset);
+          first = val + " PRECEDING";
+        } else {
+          val = Math.abs(upperOffset);
+          first = val + " FOLLOWING";
+        }
+        if (group.upperBound.isPreceding()) {
+          val = Math.abs(upperOffset);
+          second = val + " PRECEDING";
+        } else {
+          val = Math.abs(upperOffset);
+          second = val + " FOLLOWING";
+        }
+        throw InvalidInput.exception(
+            "The first value of range in the window [%s] should be lesser than the second value [%s]",
```

Comment:
I think this check should be in `DruidSqlValidator` ; it will need similar things like https://github.com/apache/druid/pull/15746